packages/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncherPlugin.java

```diff
@@ -101,6 +124,18 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
               return false;
             }
           });
+
+      // The launcher plugin sent a result receiver; send another result receiver back.
```

Comment:
This is a neat solution but I prefer using BroadcastReceivers to solve this problem. The reason is this code is quite complex (two callbacks used to accomplish a simple task). It also seems like in general we would like to control the webview activity from the plugin. BroadcastReceiver is a good solution for that. See example here: http://codesfor.in/how-to-finish-one-activity-from-another-activity-in-android/